transport/http/request.go

```diff
@@ -176,6 +177,13 @@ func (r *Request) Build(ctx context.Context) *http.Request {
 		// "unknown". This is unwanted behavior.
 		if req.ContentLength != 0 && r.stream != nil {
 			req.Body = iointernal.NewSafeReadCloser(ioutil.NopCloser(stream))
+			// Implement GetBody so the HTTP Client can retry with body.
+			req.GetBody = func() (io.ReadCloser, error) {
+				buf := new(bytes.Buffer)
```

Comment:
I think this is something we _want_ to do at a surface level but I don't want to just blindly switch to in-memory on a redirect.

At the very least I would think we can assert whether the body is a seeker such that we could rewind instead.

Beyond that I'm questioning whether we should do the memory fallback at all, or at least have it be opt-in.